pkg/cmd/roachtest/tests/cdc.go

```diff
@@ -66,247 +67,403 @@ const (
-	changefeedLogger, err := t.L().ChildLogger("changefeed")
-	if err != nil {
-		t.Fatal(err)
+func (t *cdcTester) DB() *gosql.DB {
+	return t.cluster.Conn(t.ctx, t.rt.L(), 1)
+}
+
+func (t *cdcTester) Close() {
+	t.rt.Status("cdcTester closing")
+	close(t.doneCh)
+	t.m.Wait()
+	t.logger.Close()
+	t.rt.Status("cdcTester closed")
+}
+
+type changefeedJob struct {
+	ctx            context.Context
+	sinkURI        string
+	jobID          int
+	targets        []string
+	opts           map[string]string
+	logger         *logger.Logger
+	db             *gosql.DB
+	tolerateErrors bool
+}
+
+var allTpccTargets []string = []string{
+	`tpcc.warehouse`,
+	`tpcc.district`,
+	`tpcc.customer`,
+	`tpcc.history`,
+	`tpcc.order`,
+	`tpcc.new_order`,
+	`tpcc.item`,
+	`tpcc.stock`,
+	`tpcc.order_line`,
+}
+
+var allLedgerTargets []string = []string{
+	`ledger.customer`,
+	`ledger.transaction`,
+	`ledger.entry`,
+	`ledger.session`,
+}
+
+type feedArgs struct {
+	sinkType   sinkType
+	targets    []string
+	opts       map[string]string
+	kafkaChaos bool
+	assumeRole string
+}
+
+// TODO: Maybe move away from feedArgs since its only 3 things
+func (t *cdcTester) newChangefeed(args feedArgs) changefeedJob {
+	sinkURI, ok := t.sinkCache[args.sinkType]
+	if !ok {
+		t.rt.Status(fmt.Sprintf("initiating %s sink", args.sinkType))
+		sinkURI = t.setupSink(args)
+		t.sinkCache[args.sinkType] = sinkURI
 	}
-	defer changefeedLogger.Close()
-	verifier := makeLatencyVerifier(
-		args.targetInitialScanLatency,
-		args.targetSteadyLatency,
-		changefeedLogger,
-		t.Status,
-		args.crdbChaos,
-	)
-	defer verifier.maybeLogLatencyHist()
 
-	m.Go(func(ctx context.Context) error {
-		// Some of the tests have a tight enough bound on targetSteadyLatency
-		// that the default for kv.closed_timestamp.target_duration means the
-		// changefeed is never considered sufficiently caught up. We could
-		// instead make targetSteadyLatency less aggressive, but it'd be nice to
-		// keep it where it is.
-		//
-		// TODO(ssd): As of 797819b35f5 this is actually increasing rather than decreasing
-		// the closed_timestamp.target_duration. We can probably remove this. However,
-		// as of 2021-04-20, we want to understand why this test has started failing more often
-		// before changing this.
-		if _, err := db.Exec(
-			`SET CLUSTER SETTING kv.closed_timestamp.target_duration='10s'`,
-		); err != nil {
-			t.Fatal(err)
-		}
+	targetsStr := strings.Join(args.targets, ", ")
 
-		// With a target_duration of 10s, we won't see slow span logs from changefeeds untils we are > 100s
-		// behind, which is well above the 60s targetSteadyLatency we have in some tests.
-		if _, err := db.Exec(
-			`SET CLUSTER SETTING changefeed.slow_span_log_threshold='30s'`,
-		); err != nil {
-			// We don't hard fail here because, not all versions support this setting
-			t.L().Printf("failed to set cluster setting: %s", err)
-		}
+	feedOptions := make(map[string]string)
+	feedOptions["min_checkpoint_frequency"] = "'10s'"
```

Comment:
wdyt about using changefeedbase.StatementOptions (since you're refactoring this anyway)?

Reply:
I'm not sure it ends up being helpful in this case given that this is test code where I want to optimize for "its easy to read _exactly_ whats actually happening to the statements being ran" rather than logic/values being hidden for little benefit. `StatementOptions`'s methods don't seem to be structured around "fill in default options" that I'm trying to do here, just around validation, and any errors that could be validated with the `StatementOptions` would happen immediately when you run the test anyway.

Reply:
Sure -- it's just we're "hardcoding" option names -- I thought that perhaps using explicit construction is ... well, might be better.